### ISSUE TITLE
spendable sstxchange is being detected as immature/expiry

### ISSUE THREAD

**Issue by raedah** [decred/dcrd#945]
This issue was discovered as a result of testing with https://github.com/decred/dcrwallet/pull/963

Appears it is using the 16 block period before a ticket is eligible to vote, as the measure for if the sstxchange is usable. Sstxchange is available for spending as soon as it is mined in a block, hence why it was selected as an eligible input in the first place. The error also calls it an 'expiry' of which it is not.

2017-12-10 05:53:31.672 [ERR] TKBY: Failed to purchase tickets this round: failed to send split transaction: -1: Rejected transaction xxx: tx xxx tried to spend transaction yyy including an expiry from height 192346 at height 192348 before required maturity of 16 blocks

example sstxchange here
https://testnet.dcrdata.org/explorer/tx/389bd79e964431a6ca2576cce2c9abe777c8c34b416ddd2719cb2a8a823e2a1e

https://github.com/decred/dcrd/blob/b13f5dfbdac0b3ea654b082ad38cd134613920be/blockchain/validate.go#L1711

**Comment by jrick** [comment]
This is consensus code and is intentional to prevent the transaction from being invalidated due to the expiry requirement across a reorg, wiping out the transaction chain in the process.  The maturity period is always the coinbase maturity period, since the same problem can affect coinbases and is why they have a maturity.

Instead what needs to happen is wallet must not include these immature outputs in the spendable utxo set.

Because we used expiry with ticket purchases to prevent them from lingering too long when there was much competition for block inclusion, this maturity period also affected the ticket purchase change, and is precisely why we went with the split transactions.
